libs/ng-lib/src/lib/transfer-state/transfer-state.service.ts

```diff
@@ -57,11 +57,11 @@ export class TransferStateService {
     /** prevent emitting before navigation to _this_ URL is done. */
     switchMap((e: NavigationStart) =>
       this.router.events.pipe(
-        filter((ev) => ev instanceof NavigationEnd && ev.url === e.url),
+        filter((ev) => ev instanceof GuardsCheckEnd && ev.url === e.url),
```

Comment:
Should we add a test around this for future purposes that when we change it to another `instance` we have a failing test?

Reply:
That is actually a good idea. I'll add a route that uses transferstae in a resolver to our kitchen sick.